src/phy/mod.rs

```diff
@@ -132,6 +137,33 @@ pub use self::tracer::Tracer;
 ))]
 pub use self::tuntap_interface::TunTapInterface;
 
+/// An ID that can be used to uniquely identify a packet to a [`Device`],
+/// sent or received by that same [`Device`]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Debug, PartialEq)]
+pub struct PacketId {
+    id: usize,
+}
```

Comment:
Maybe we could write it as:
```rust
pub struct PacketId(usize);
```

Reply:
Yes.

Personally, I'm a bigger fan of structs with named fields but it seems much more in line with the rest of the lib (and it's probably more idiomatic?) to use a unit struct here. I'll change it.